src/params/types.ts

```diff
@@ -28,6 +30,19 @@
 export abstract class Expression<T extends string | number | boolean | string[]> {
   // Returns the Expression's runtime value, based on the CLI's resolution of params.
   value(): T {
+    if (process.env.FIREBASE_DISCOVERY_DIR) {
+      logger.warn(
+        `Parametrized expression ${this.toString()}.value() invoked during Function discovery. This is usually a mistake.`
+      );
+      logger.warn(
+        `To configure a Function with an expression object, use it directly without calling .value().`
+      );
+    }
+    return this.runtimeValue();
+  }
+
+  // @internal
```

Comment:
Do we need @internal tag on all of implementation of `runtimeValue`?

Reply:
Possibly, and while I was down the rabbit hole of looking that up, I also realized that tsc doesn't respect jsdoc annotations unless they're in, well, jsdoc format, so fixing that too.